NitroxServer/GameLogic/PlayerManager.cs

```diff
@@ -70,14 +71,16 @@ public List<Player> GetPlayers()
             }
         }
 
-        public Player CreatePlayer(Connection connection, string reservationKey)
+        public Player CreatePlayer(Connection connection, string reservationKey, out bool wasBrandNewPlayer)
         {
             lock (assetsByConnection)
             {
                 ConnectionAssets assetPackage = assetsByConnection[connection];
                 PlayerContext playerContext = reservations[reservationKey];
                 Validate.NotNull(playerContext);
 
+                wasBrandNewPlayer = playerContext.WasBrandNewPlayer;
```

Comment:
do you need to pass it through the playerContext? can you simply do` wasBrandNewPlayer = playerData.hasSeenPlayerBefore(playerName);?`

Reply:
I have to, because by that time the player is created in playersByPlayerName and when I call it it's the last chance and I have to somehow get it to CreatePlayer.